src/drivers/distance_sensor/lightware_laser_i2c/lightware_laser_i2c.cpp

```diff
@@ -386,6 +399,39 @@ void LightwareLaser::start()
 	ScheduleDelayed(_conversion_interval);
 }
 
+bool LightwareLaser::is_restricted()
+{
+	px4::msg::VehicleStatus vehicle_status;
+
+	if (_vehicle_status_sub.update(&vehicle_status)) {
+		// Check if vehicle type changed
+		if (vehicle_status.vehicle_type != _vehicle_type) {
+			// Transition VTOL -> Fixed Wing
+			if (_vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_ROTARY_WING &&
+			    vehicle_status.vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_FIXED_WING) {
+				_restriction = true;
+			}
+
+			// Transition Fixed Wing -> VTOL
+			else if (_vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_FIXED_WING &&
+				 vehicle_status.vehicle_type == px4::msg::VehicleStatus::VEHICLE_TYPE_ROTARY_WING) {
+				_restriction = false;
+			}
+
+			_vehicle_type = vehicle_status.vehicle_type;
+		}
+	}
+
+	updateParams();
+	if (_param_mode.get() == 0) {
+		return true;
+	}
+	if (_param_mode.get() == 1) {
+		return false;
+	}
```

Comment:
Maybe a switch case for the parameter values where if it's not 2 the entire subscription doesn't even do work.